processing/src/main/java/io/druid/query/QueryInterruptedException.java

```diff
@@ -22,27 +22,87 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.TimeoutException;
+
 public class QueryInterruptedException extends RuntimeException
 {
-  public QueryInterruptedException() {
-    super();
-  }
+  public static final String QUERY_INTERRUPTED = "Query interrupted";
+  public static final String QUERY_TIMEOUT = "Query timeout";
+  public static final String QUERY_CANCELLED = "Query cancelled";
+  public static final String UNKNOWN_EXCEPTION = "Unknown exception";
+
+  private static final List<String> listKnownException = Arrays.asList(
```

Comment:
Given that the point of this data structure is to call `.contains()` it would be a lot better as a `Set`.